tardis/io/atomic.py

```diff
@@ -310,51 +128,85 @@ def from_hdf5(cls, fname=None):
         if not os.path.exists(fname):
             raise ValueError("Supplied Atomic Model Database %s does not exists" % fname)
 
-        atom_data = read_basic_atom_data(fname)
-        ionization_data = read_ionization_data(fname)
-        levels_data = read_levels_data(fname)
-        lines_data = read_lines_data(fname)
-
-        with h5py.File(fname, 'r') as h5_file:
-            h5_datasets = h5_file.keys()
-
-        if 'macro_atom_data' in h5_datasets:
-            macro_atom_data = read_macro_atom_data(fname)
-        else:
-            macro_atom_data = None
-
-        if 'zeta_data' in h5_datasets:
-            zeta_data = read_zeta_data(fname)
-        else:
-            zeta_data = None
-
-        if 'collision_data' in h5_datasets:
-            collision_data, collision_data_temperatures = read_collision_data(fname)
-        else:
-            collision_data, collision_data_temperatures = (None, None)
-
-        if 'synpp_refs' in h5_datasets:
-            synpp_refs = read_synpp_refs(fname)
-        else:
-            synpp_refs = None
-
-        if 'ion_cx_data' in h5_datasets and 'ion_cx_data' in h5_datasets:
-            ion_cx_data = read_ion_cx_data(fname)
-        else:
-            ion_cx_data = None
-
-        atom_data = cls(atom_data=atom_data, ionization_data=ionization_data, levels_data=levels_data,
-                        lines_data=lines_data, macro_atom_data=macro_atom_data, zeta_data=zeta_data,
-                        collision_data=(collision_data, collision_data_temperatures), synpp_refs=synpp_refs,
-                        ion_cx_data=ion_cx_data)
-
-        with h5py.File(fname, 'r') as h5_file:
-            atom_data.uuid1 = h5_file.attrs['uuid1']
-            atom_data.md5 = h5_file.attrs['md5']
-            atom_data.version = h5_file.attrs.get('database_version', None)
-
-            if atom_data.version is not None:
-                atom_data.data_sources = pickle.loads(h5_file.attrs['data_sources'])
+        with pd.HDFStore(fname) as store:
```

Comment:
Yes, I know. But it would be possible, to go around that. For the `collision data`, one could simply add the unit straight away. It makes far more sense for `__init__` to accept a dataframe instead of a tuple.